src/storage/txn/task.rs

```diff
@@ -107,8 +159,8 @@ mod tests {
     fn test_alloc_memory_quota() {
         let p = PrewriteRequest::default();
         let cmd: TypedCommand<_> = p.into();
-        let mut task = Task::new(0, cmd.cmd);
         let quota = Arc::new(MemoryQuota::new(1 << 32));
+        let mut task = Task::allocate(0, cmd.cmd, quota.clone()).unwrap();
         task.alloc_memory_quota(quota.clone()).unwrap();
```

Comment:
Is it allocated twice?

Reply:
If we guarantee that it must have successfully allocated from the quota once created, it looks that the old `alloc_memory_quota` function becomes not necessary.

Reply:
The test is refined.